libs/community/langchain_community/tools/azure_ai_services/azure_document_translation.py

```
@@ -0,0 +1,178 @@
+        loader = loader_class(file_path)
+        data = loader.load()
+
+        return " ".join([doc.page_content for doc in data])
+
+    def read_text(self, file_path: str) -> str:
+        """Read text from a plain text file."""
+        with open(file_path, "r", encoding="utf-8") as file:
+            return file.read().strip()
+
+    def translate_text(self, text: str, target_language: Optional[str] = None) -> str:
+        """
+        Translate the input text to the target language
+        using the Azure Text Translation API.
+
+        Args:
+            text (str): The text to be translated.
+            target_language (str, optional):
+                The target language for translation (default: Spanish).
+
+        Returns:
+            str: Translated text.
+
+        Raises:
+            RuntimeError: If the translation request fails.
+        """
+        if target_language is None:
+            target_language = self.target_language
+
+        try:
+            from azure.ai.translation.text.models import InputTextItem
+        except ImportError:
+            raise ImportError(
+                "Run 'pip install azure-ai-translation-text'.")
+
+        try:
+            request_body = [InputTextItem(text=text)]
+            response = self.client.translate(content=request_body, to=[target_language])
+
+            translations = response[0].translations
+            if translations:
+                return translations[0].text
+            return ""  # No translations found
+        except Exception as e:
+            logger.error(f"Error occurred during translation: {e}")
+            raise
+
+    def _run(self, query: str,
+             run_manager: Optional[CallbackManagerForToolRun] = None) -> str:
+        """"Run the tool"""
+        try:
+            return self.translate_text(query)
+        except Exception as e:
+            logger.error(f"Error while running AzureDocumentTranslateTool: {e}")
```

I think we can just raise the exception in this case and remove the logging